cdap-ui/app/cdap/services/helpers.js

```diff
@@ -215,11 +215,23 @@ function getArtifactNameAndVersion(nameWithVersion) {
   }
   let regExpRule = new RegExp('\\-(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:[.\\-](.*))?$');
   let version = regExpRule.exec(nameWithVersion);
+  if (version && Array.isArray(version)) {
+    version = version[0].slice(1);
+  } else {
+    // when version is the filename i.e 1.2.3.jar
+    let nameIsVersionRegEx = new RegExp('(\\d+)(?:\\.(\\d+))?(?:\\.(\\d+))?(?:[.\\-](.*))?$');
+    let validVersion = nameIsVersionRegEx.exec(nameWithVersion);
+    if (validVersion && Array.isArray(validVersion)) {
+      return { name: nameWithVersion, version: nameWithVersion };
+    }
+  }
+  let name = version
+    ? nameWithVersion.substr(0, nameWithVersion.indexOf(version) - 1)
+    : nameWithVersion;
+  // if version is not present, default it to 1.0.0-SNAPSHOT
   if (!version) {
-    return { name: nameWithVersion, version: undefined };
+    version = '1.0.0-SNAPSHOT';
```

Comment:
this should be 1.0.0, without the SNAPSHOT

Reply:
fixed.